src/main/java/uk/gov/companieshouse/authcodenotification/controller/AuthCodeNotificationController.java

```diff
@@ -63,6 +64,8 @@ public ResponseEntity<Object> sendEmail(@RequestHeader(value = ERIC_REQUEST_ID_K
 
         try {
             authCodeNotificationService.sendAuthCodeEmail(requestId, authCode, companyNumber);
+        } catch (EntityNotFoundException nfe) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
```

Comment:
We should be logging the exception details at this point. In the catch block below, too.